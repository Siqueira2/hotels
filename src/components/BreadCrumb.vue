<template>
  <q-breadcrumbs class="text-grey-6">
    <template v-slot:separator>
      <q-icon size="1.5em" name="chevron_right" color="primary" />
    </template>

    <q-breadcrumbs-el :label="$t('shared.breadcrumbs.home')" to="/" />
    <q-breadcrumbs-el :label="$t('shared.breadcrumbs.hotels')" />
    <template v-if="breadcrumbs">
      <q-breadcrumbs-el :label="breadcrumbs" />
    </template>
  </q-breadcrumbs>
</template>

<script setup lang="ts">
import { computed } from 'vue'
import { useRoute } from 'vue-router'
import { useI18n } from 'vue-i18n'

const { t } = useI18n()
const route = useRoute()

const breadcrumbs = computed(() =>
  t('shared.breadcrumbs.accommodation', { suffix: route.query.name }),
)
</script>
