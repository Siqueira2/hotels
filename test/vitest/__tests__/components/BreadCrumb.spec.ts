import { describe, it, expect, vi, beforeEach } from 'vitest'
import type { VueWrapper } from '@vue/test-utils'
import { mount } from '@vue/test-utils'

import BreadCrumb from 'src/components/BreadCrumb.vue'

vi.mock('vue-router', () => {
  return { useRoute: () => ({ query: { name: 'Belo Horizonte' } }) }
})

const createWrapper = () =>
  mount(BreadCrumb, {
    global: { mocks: { $router: { push: vi.fn() } } },
  }) as VueWrapper

describe('BreadCrumb', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = createWrapper()
  })

  it('should match snapshots', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should contain correct text', () => {
    expect(wrapper.text()).toContain('Belo Horizonte')
  })
})
