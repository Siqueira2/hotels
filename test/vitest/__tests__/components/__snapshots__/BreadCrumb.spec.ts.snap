// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BreadCrumb > should match snapshots 1`] = `
"<div class=\\"q-breadcrumbs text-grey-6\\">
  <div class=\\"flex items-center justify-start q-gutter-sm\\">
    <div class=\\"flex items-center text-primary\\"><span class=\\"q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable\\">shared.breadcrumbs.home</span></div>
    <div class=\\"q-breadcrumbs__separator\\"><i class=\\"q-icon text-primary notranslate material-icons\\" style=\\"font-size: 1.5em;\\" aria-hidden=\\"true\\" role=\\"presentation\\">chevron_right</i></div>
    <div class=\\"flex items-center text-primary\\"><span class=\\"q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable\\">shared.breadcrumbs.hotels</span></div>
    <div class=\\"q-breadcrumbs__separator\\"><i class=\\"q-icon text-primary notranslate material-icons\\" style=\\"font-size: 1.5em;\\" aria-hidden=\\"true\\" role=\\"presentation\\">chevron_right</i></div>
    <div class=\\"flex items-center q-breadcrumbs--last\\"><span class=\\"q-breadcrumbs__el q-link flex inline items-center relative-position q-link--focusable\\">Hospedagem em Belo Horizonte</span></div>
  </div>
</div>"
`;
